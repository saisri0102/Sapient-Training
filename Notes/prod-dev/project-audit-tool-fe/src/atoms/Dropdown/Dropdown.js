import React from "react";
import PropTypes from "prop-types";
import { DropdownStyle } from "./Dropdown.style";
import withStyles from "../../withstyle";

const Dropdown = (props) => {
  const {
    name,
    varient,
    options,
    onChange,
    className,
    labelname,
    labelclassName,
    defaultValue,
    ...otherProps
  } = props;

  return (
    <span varient={varient} className={`${className}`}>
      <label htmlFor={name} className="dropdown-label">
        {labelname}
      </label>
      <select
        className="dropdown-select"
        onChange={onChange}
        value={defaultValue}
        name={name}
        {...otherProps}
      >
        {options.map((option) => (
          <option value={option.value} selected={option.status}>
            {option.label}
          </option>
        ))}
      </select>
    </span>
  );
};

Dropdown.propTypes = {
  name: PropTypes.string,
  options: PropTypes.array,
  onChange: PropTypes.func,
  labelname: PropTypes.string,
  labelclassName: PropTypes.string,
  className: PropTypes.string,
};

Dropdown.defaultProps = {
  options: [
    {
      label: "Select a Category",
      value: "Select a Category",
    },
  ],
  onChange: () => console.log("Please enter the onChange info"),

  className: "",
  labelname: "",
  labelclassName: "",
};

export default withStyles(Dropdown, DropdownStyle);
export { Dropdown as DropdownNotStyled };
