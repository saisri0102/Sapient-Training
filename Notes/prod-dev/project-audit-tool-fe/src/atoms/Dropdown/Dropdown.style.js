import styled, { css } from "styled-components";

export const DropdownStyle = css`
  .dropdown-label {
    ${(props) => {
      switch (props.labelclassName) {
        case "orderbylabel":
          return css`
            color:  #777777; 
          `;
        case "sortbylabel":
          return css`
            color:  #777777; 
          `;

        default:
          break;
      }
    }}
  }
  .dropdown-select {
    ${(props) => {
      switch (props.className) {
        case "order":
          return css`
            background-color: inherit;
            border: none;
            font-weight: bold;
            font-size: 1em;
          `;
        case "sort":
          return css`
            background-color: inherit;
            border: none;
            font-weight: bold;
            font-size: 1em;
          `;

        case "category-form":
          return css`
            width: 100%;
            background-color:inherit;
            border: none;
            border-bottom: 1px solid
              ${(props) => {
                try {
                  return props.theme.colors.darkGreyColor;
                } catch {
                  return null;
                }
              }};
          `;

        case "category-sm":
          return css`
            width: 150px;
            border: none;
            border-bottom: 1px solid
              ${(props) => {
                try {
                  return props.theme.colors.darkGreyColor;
                } catch {
                  return null;
                }
              }};
          `;
        default:
          break;
      }
    }}
  }
`;

// export const LABEL = styled.label`
//   ${(props) => {
//     switch (props.className) {
//       case "orderbylabel":
//         return css`
//           color:${(props) => {
//             try {
//                 return props.theme.colors.darkGreyColor;
//             } catch {
//                 return null;
//             }
//          }};
//         `;
//       case "sortbylabel":
//         return css`
//           color: ${(props) => {
//             try {
//                 return props.theme.colors.darkGreyColor;
//             } catch {
//                 return null;
//             }
//          }};
//           margin-left: 5px;
//         `;

//       default:
//         break;
//     }
//   }}
// `;
// export const SELECT = styled.select`
//   ${(props) => {
//     switch (props.className) {
//       case "order":
//         return css`
//           border: none;
//           font-weight: bold;
//           font-size:1em;
//         `;
//       case "sort":
//         return css`
//           border: none;
//           font-weight: bold;
//           font-size:1em;
//         `;

//       case "category-form":
//         return css`
//           width: 100%;
//           border: none;
//           border-bottom: 1px solid ${(props) => {
//             try {
//                 return props.theme.colors.darkGreyColor;
//             } catch {
//                 return null;
//             }
//          }};
//         `;

//       case "category-sm":
//         return css`
//           width: 150px;
//           border: none;
//           border-bottom: 1px solid ${(props) => {
//             try {
//                 return props.theme.colors.darkGreyColor;
//             } catch {
//                 return null;
//             }
//          }};
//         `;
//       default:
//         break;
//     }
//   }}
// `;
