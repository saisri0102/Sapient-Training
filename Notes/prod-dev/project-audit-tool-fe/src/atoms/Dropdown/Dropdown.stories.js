import React from "react";
import { storiesOf } from "@storybook/react";

import Dropdown from "./Dropdown"; 

storiesOf("Dropdown", module)
  .add("Order", () => (
    <Dropdown
      className="order"
      labelclassName="orderbylabel"
      labelname="Order By: "
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "ASC",
          value: "ascending",
        },
        {
          label: "DSC",
          value: "descending",
        },
      ]}
    />
  ))
  .add("Sort", () => (
    <Dropdown
      className="sort"
      labelclassName="sortbylabel"
      labelname="Sort By: "
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Name",
          value: "name",
        },
        {
          label: "Author",
          value: "author",
        },
        {
          label: "Progress",
          value: "progress",
        },
      ]}
    />
  ))
  .add("Category-Large", () => (
    <Dropdown
      className="category-form"
      labelclassName=""
      labelname=""
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Select a Category",
          value: "0",
        },
        {
          label: "Mango",
          value: "mango",
        },
        {
          label: "Banana",
          value: "banana",
        },
        {
          label: "Pineapple",
          value: "pineapple",
        },
      ]}
    />
  ))
  .add("CategorySmall", () => (
    <Dropdown
      className="category-sm"
      abelclassName=""
      labelname=""
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Select a Category",
          value: "0",
        },
        {
          label: "Mango",
          value: "mango",
        },
        {
          label: "Banana",
          value: "banana",
        },
        {
          label: "Pineapple",
          value: "pineapple",
        },
      ]}
    />
  ));
