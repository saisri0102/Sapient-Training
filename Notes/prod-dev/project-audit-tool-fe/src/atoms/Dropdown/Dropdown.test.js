import {DropdownNotStyled} from "./Dropdown";

import { render, screen } from "@testing-library/react";

import userEvent from "@testing-library/user-event";

test("Order Dropdown", () => { 
  const wrapper = render(
    <DropdownNotStyled
      className="order"
      labelclassName="orderbylabel"
      labelname="Order By: "
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "ASC",
          value: "ascending",
        },
        {
          label: "DSC",
          value: "descending",
        },
      ]}
    />
  );

  expect(wrapper).toBeDefined();

  expect(wrapper).toMatchSnapshot();
});

test("Sort Dropdown", () => {
  const wrapper = render(
    <DropdownNotStyled
      className="sort"
      labelclassName="sortbylabel"
      labelname="Sort By: "
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Name",
          value: "name",
        },
        {
          label: "Author",
          value: "author",
        },
        {
          label: "Progress",
          value: "progress",
        },
      ]}
    />
  );

  expect(wrapper).toBeDefined();

  expect(wrapper).toMatchSnapshot();
});

test("Category Large", () => {
  const wrapper = render(
    <DropdownNotStyled
      className="category-form"
      labelclassName=""
      labelname=""
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Select a Category",
          value: "0",
        },
        {
          label: "Mango",
          value: "mango",
        },
        {
          label: "Banana",
          value: "banana",
        },
        {
          label: "Pineapple",
          value: "pineapple",
        },
      ]}
    />
  );

  expect(wrapper).toBeDefined();

  expect(wrapper).toMatchSnapshot();
});

test("Category Small", () => {
  const wrapper = render(
    <DropdownNotStyled
      className="category-sm"
      abelclassName=""
      labelname=""
      onChange={() => {
        console.log("Onchange Event");
      }}
      options={[
        {
          label: "Select a Category",
          value: "0",
        },
        {
          label: "Mango",
          value: "mango",
        },
        {
          label: "Banana",
          value: "banana",
        },
        {
          label: "Pineapple",
          value: "pineapple",
        },
      ]}
    />
  );

  expect(wrapper).toBeDefined();

  expect(wrapper).toMatchSnapshot();
});
