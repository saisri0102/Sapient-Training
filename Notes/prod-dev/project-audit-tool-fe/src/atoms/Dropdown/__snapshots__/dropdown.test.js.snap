// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Large 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-gsDJrp jWQrnZ"
      >
        
        <select
          class="sc-bdvvaa hGcuLD category-form"
        >
          <option
            value="0"
          >
            Select a Category
          </option>
          <option
            value="mango"
          >
            Mango
          </option>
          <option
            value="banana"
          >
            Banana
          </option>
          <option
            value="pineapple"
          >
            Pineapple
          </option>
        </select>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-gsDJrp jWQrnZ"
    >
      
      <select
        class="sc-bdvvaa hGcuLD category-form"
      >
        <option
          value="0"
        >
          Select a Category
        </option>
        <option
          value="mango"
        >
          Mango
        </option>
        <option
          value="banana"
        >
          Banana
        </option>
        <option
          value="pineapple"
        >
          Pineapple
        </option>
      </select>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Category Small 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-gsDJrp jWQrnZ"
      >
        
        <select
          class="sc-bdvvaa eJinxh category-sm"
        >
          <option
            value="0"
          >
            Select a Category
          </option>
          <option
            value="mango"
          >
            Mango
          </option>
          <option
            value="banana"
          >
            Banana
          </option>
          <option
            value="pineapple"
          >
            Pineapple
          </option>
        </select>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-gsDJrp jWQrnZ"
    >
      
      <select
        class="sc-bdvvaa eJinxh category-sm"
      >
        <option
          value="0"
        >
          Select a Category
        </option>
        <option
          value="mango"
        >
          Mango
        </option>
        <option
          value="banana"
        >
          Banana
        </option>
        <option
          value="pineapple"
        >
          Pineapple
        </option>
      </select>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Order Dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-gsDJrp kUqObB orderbylabel"
      >
        Order By: 
        <select
          class="sc-bdvvaa dmVvba order"
        >
          <option
            value="ascending"
          >
            ASC
          </option>
          <option
            value="descending"
          >
            DSC
          </option>
        </select>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-gsDJrp kUqObB orderbylabel"
    >
      Order By: 
      <select
        class="sc-bdvvaa dmVvba order"
      >
        <option
          value="ascending"
        >
          ASC
        </option>
        <option
          value="descending"
        >
          DSC
        </option>
      </select>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Sort Dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-gsDJrp llwOr sortbylabel"
      >
        Sort By: 
        <select
          class="sc-bdvvaa jlwfWE sort"
        >
          <option
            value="name"
          >
            Name
          </option>
          <option
            value="author"
          >
            Author
          </option>
          <option
            value="progress"
          >
            Progress
          </option>
        </select>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-gsDJrp llwOr sortbylabel"
    >
      Sort By: 
      <select
        class="sc-bdvvaa jlwfWE sort"
      >
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="author"
        >
          Author
        </option>
        <option
          value="progress"
        >
          Progress
        </option>
      </select>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
